From 0ada7bd2016c196215c4a391cc9c5f7655c34500 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Oliver=20St=C3=A4bler?= <oliver.staebler@bytesatwork.ch>
Date: Mon, 8 Mar 2021 15:39:18 +0100
Subject: [PATCH] arm: dts: stm32mp157c-bytedevkit: Support different RAM size
 configurations

Select the correct RAM size and definitions according to
CONFIG_BYTEENGINE_M5_RAM.
---
 arch/arm/dts/stm32mp157c-bytedevkit-u-boot.dtsi |  7 +++++++
 arch/arm/dts/stm32mp157c-byteengine-m5.dtsi     | 10 +++++++++-
 2 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/stm32mp157c-bytedevkit-u-boot.dtsi b/arch/arm/dts/stm32mp157c-bytedevkit-u-boot.dtsi
index 8a6f437a93..0ed92e982d 100644
--- a/arch/arm/dts/stm32mp157c-bytedevkit-u-boot.dtsi
+++ b/arch/arm/dts/stm32mp157c-bytedevkit-u-boot.dtsi
@@ -5,7 +5,14 @@
 
 #include <dt-bindings/clock/stm32mp1-clksrc.h>
 #include "stm32mp15-u-boot.dtsi"
+
+#if defined(CONFIG_BYTEENGINE_STM32MP1_RAM_512)
 #include "stm32mp15-ddr3-1x4Gb-1066-binG.dtsi"
+#elif defined(CONFIG_BYTEENGINE_STM32MP1_RAM_1024)
+#include "stm32mp15-ddr3-1x8Gb-1066-binG.dtsi"
+#else
+#error No RAM configuration for byteengine defined!
+#endif
 
 / {
 	aliases {
diff --git a/arch/arm/dts/stm32mp157c-byteengine-m5.dtsi b/arch/arm/dts/stm32mp157c-byteengine-m5.dtsi
index 30259b58d0..b0ad337709 100644
--- a/arch/arm/dts/stm32mp157c-byteengine-m5.dtsi
+++ b/arch/arm/dts/stm32mp157c-byteengine-m5.dtsi
@@ -11,11 +11,19 @@
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/rtc/rtc-stm32.h>
 
+#if defined(CONFIG_BYTEENGINE_STM32MP1_RAM_512)
+#define STM32MP1_RAM_SIZE 0x20000000
+#elif defined(CONFIG_BYTEENGINE_STM32MP1_RAM_1024)
+#define STM32MP1_RAM_SIZE 0x40000000
+#else
+#error No RAM configuration for byteengine defined!
+#endif
+
 / {
 	compatible = "batw,m5", "st,stm32mp157";
 
 	memory@c0000000 {
-		reg = <0xc0000000 0x20000000>;
+		reg = <0xc0000000 STM32MP1_RAM_SIZE>;
 	};
 
 	reserved-memory {
-- 
2.20.1

